.gitignore

```diff
@@ -54,3 +54,7 @@ config.local.js
 
 build_output
 tmp
+
+src/js/built-temp
+translations/tmp
+gulp/additional_build_files
```

Comment:
`tmp` already matches `translations/tmp` (remove the former?); also maybe this stuff should be rearranged to be by the `# Buildfiles` stuff above

Reply:
As I understand, the root .gitignore was copied from GitHub templates for Node.js project. There are many redundant patterns, and I didn't touch any of these.

Reply:
I guess in this aspect this PR is just rearranging things; the temp files' locations need to be revisited in the future probbaly